azurerm/internal/services/compute/resource_arm_managed_disk.go

```diff
@@ -144,7 +145,7 @@ func resourceArmManagedDisk() *schema.Resource {
 				// TODO: make this case-sensitive once this bug in the Azure API has been fixed:
 				//       https://github.com/Azure/azure-rest-api-specs/issues/8132
 				DiffSuppressFunc: suppress.CaseDifference,
-				ValidateFunc:     azure.ValidateResourceID,
+				ValidateFunc:     validate.DiskEncryptionSetID,
```

Comment:
unfortunately we can't use this here until the API returns this in the correct case

Reply:
Done

Reply:
@ArcturusZhang I think you may have forgotten to push that commit? :)

Reply:
😂how silly of me